xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/migration/R141004000XWIKI20575DataMigration.java

```diff
@@ -92,9 +104,20 @@ public XWikiDBVersion getVersion()
     public void migrate() throws DataMigrationException
     {
         try {
-            this.solrInstance.deleteByQuery("*:*");
-            this.solrInstance.commit();
-        } catch (SolrServerException | IOException e) {
+            Map<String, Map<String, Object>> fields =
+                this.solrSchemaUtils.getFields(SolrClientInstance.CORE_NAME, true);
+
+            // We only check for the presence of the title__ field, which should be representative of the dynamic field
+            if (!fields.containsKey("title__")) {
+                this.solrInstance.deleteByQuery("*:*");
```

Comment:
I have a doubt about that logic: if it's a remote solr then the schema would be updated by the new config file, but I'd still need to perform the reindex no? Meaning that I need to move this call outside the if?

Reply:
Yes, I agree.

Reply:
I fixed it.